libmamba/src/api/install.cpp

```diff
@@ -656,12 +656,33 @@ namespace mamba
                 }
             }
 
+            static std::unique_ptr<TemporaryFile> tmp_lock_file;
```

Comment:
Why `static`? this makes the feature unusable in a library context since it will be destroyed at the program exit only.

Reply:
in the micromamba context, if it gets destroyed before, the file is gone. Let's look for a better solution :)